php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1498 @@
+         b.optional(b.token(AMPERSAND)),
+         VARIABLE_IDENTIFIER()
+     ));
+  }
+  
+  public ExpressionTree STATIC_SCALAR() {
+    // FIXME (PY) : In the old grammar: b.firstOf(COMBINED_SCALAR, CONDITIONAL_EXPR)
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.STATIC_SCALAR).is(
+      b.firstOf(
+        COMMON_SCALAR(),
+        EXPRESSION()));
+  }
+
+  public FunctionCallTree INTERNAL_FUNCTION() {
+    return b.<FunctionCallTree>nonterminal(PHPLexicalGrammar.INTERNAL_FUNCTION)
+      .is(b.firstOf(
+        f.internalFunction1(b.token(PHPLexicalGrammar.ISSET), b.token(LPARENTHESIS), EXPRESSION(), b.zeroOrMore(f.newTuple15(b.token(COMMA), EXPRESSION())), b.token(RPARENTHESIS)),
+
+        f.internalFunction2(
+          b.firstOf(
+            b.token(PHPLexicalGrammar.EMPTY),
+            b.token(PHPLexicalGrammar.EVAL)),
+          b.token(LPARENTHESIS), EXPRESSION(), b.token(RPARENTHESIS)),
+
+        f.internalFunction3(
+          b.firstOf(
+            b.token(PHPLexicalGrammar.INCLUDE_ONCE),
+            b.token(PHPLexicalGrammar.INCLUDE),
+            b.token(PHPLexicalGrammar.REQUIRE_ONCE),
+            b.token(PHPLexicalGrammar.REQUIRE),
+            b.token(PHPLexicalGrammar.CLONE),
+            b.token(PHPLexicalGrammar.PRINT)),
+          EXPRESSION())
+      ));
+  }
+
+  public ArrayInitializerTree ARRAY_INITIALIZER() {
+    return b.<ArrayInitializerTree>nonterminal(PHPLexicalGrammar.COMBINED_SCALAR)
+      .is(b.firstOf(
+        f.newArrayInitFunction(b.token(ARRAY), b.token(LPARENTHESIS), b.optional(ARRAY_PAIR_LIST()), b.token(RPARENTHESIS)),
+        f.newArrayInitBracket(b.token(LBRACKET), b.optional(ARRAY_PAIR_LIST()), b.token(RBRACKET))
+      ));
+
+  }
+
+  public SeparatedList<ArrayPairTree> ARRAY_PAIR_LIST() {
+    return b.<SeparatedList<ArrayPairTree>>nonterminal(PHPLexicalGrammar.ARRAY_PAIR_LIST)
+      .is(f.arrayInitializerList(
+        ARRAY_PAIR(),
+        b.zeroOrMore(f.newTuple17(b.token(COMMA), ARRAY_PAIR())),
+        b.optional(b.token(COMMA))
+      ));
+  }
+
+  public ArrayPairTree ARRAY_PAIR() {
+    return b.<ArrayPairTree>nonterminal(Kind.ARRAY_PAIR)
+      .is(b.firstOf(
+        f.arrayPair1(EXPRESSION(), b.optional(f.newTuple13(b.token(DOUBLEARROW), b.firstOf(REFERENCE_VARIABLE(), EXPRESSION())))),
+        f.arrayPair2(REFERENCE_VARIABLE())
+      ));
+  }
+
+  public FunctionExpressionTree FUNCTION_EXPRESSION() {
+    return b.<FunctionExpressionTree>nonterminal(Kind.FUNCTION_EXPRESSION)
+      .is(f.functionExpression(
+        b.optional(b.token(STATIC)),
+        b.token(FUNCTION),
+        b.optional(b.token(AMPERSAND)),
+        PARAMETER_LIST(),
+        b.optional(LEXICAL_VARIABLES()),
+        BLOCK()
+      ));
+
+  }
+
+  public NewExpressionTree NEW_EXPRESSION() {
+    return b.<NewExpressionTree>nonterminal(Kind.NEW_EXPRESSION)
+      .is(f.newExpression(b.token(NEW), MEMBER_EXPRESSION()));
+  }
+
+  public ExitTree EXIT_EXPRESSION() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename this method name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00100)